src/android/CodePushReportingManager.java

```diff
@@ -44,26 +22,22 @@ public CodePushReportingManager(Activity cordovaActivity, CodePushPreferences co
     /**
      * Invokes the window.codePush.reportStatus JS function for the given webView.
      */
-    public void reportStatus(Status status, String label, String appVersion, String deploymentKey, final CordovaWebView webView) {
+    public void reportStatus(StatusReport statusReport, final CordovaWebView webView) {
         /* JS function to call: window.codePush.reportStatus(status: number, label: String, appVersion: String, currentDeploymentKey: String, previousLabelOrAppVersion?: string, previousDeploymentKey?: string) */
-        if (deploymentKey == null || deploymentKey.isEmpty()) {
+        if (statusReport.deploymentKey == null || statusReport.deploymentKey.isEmpty()) {
             return;
         }
 
         final String script = String.format(
             Locale.US,
             "javascript:document.addEventListener(\"deviceready\", function () { window.codePush.reportStatus(%d, %s, %s, %s, %s, %s); });",
-            status.getValue(),
-            convertStringParameter(label),
-            convertStringParameter(appVersion),
-            convertStringParameter(deploymentKey),
-            convertStringParameter(codePushPreferences.getLastVersionLabelOrAppVersion()),
-            convertStringParameter(codePushPreferences.getLastVersionDeploymentKey())
-            );
-
-        if (status == Status.STORE_VERSION || status == Status.UPDATE_CONFIRMED) {
-            codePushPreferences.saveLastVersion(label == null ? appVersion : label, deploymentKey);
-        }
+            statusReport.status.getValue(),
+            convertStringParameter(statusReport.label),
+            convertStringParameter(statusReport.appVersion),
+            convertStringParameter(statusReport.deploymentKey),
+            statusReport.lastVersionLabelOrAppVersion == null ? convertStringParameter(codePushPreferences.getLastVersionLabelOrAppVersion()) : convertStringParameter(statusReport.lastVersionLabelOrAppVersion),
```

Comment:
why would this be null, and why would it be correct to report if the version stored in codePushPreferences if it is?

both this and the line scare me in terms of an "we could get a really cryptic error due to reporting the wrong version or reporting when we shouldn't and i never want to deal with something like that"-kind of way